client/src/container/App/App.js

```diff
@@ -5,6 +5,7 @@ import Main from '../Main/Main';
 import Footer from '../../components/Footer/Footer';
 import ErrorBoundary from '../../components/ErrorBoundary/ErrorBoundary';
 
+
 class App extends Component {
```

Comment:
Though your setup works, usually it's more convenient to define Routes in one place in your app, separating the concerns of Routing and Functionality.
E.x. you can have your Routes setup in App.js, they'll render component in which you define all the app related logic, be it conditional Header or conditional Main content.